src/workerd/jsg/jsg-test.c++

```diff
@@ -360,10 +360,54 @@ KJ_TEST("Test JSG_CALLABLE") {
   e.expectEval("let obj = getCallable(); new obj();", "boolean", "true");
 }
 
-}  // namespace
-
 // ========================================================================================
+struct InterceptContext: public ContextGlobalObject {
+  struct ProxyImpl: public jsg::Object,
+                    public jsg::NamedIntercept {
+    static jsg::Ref<ProxyImpl> constructor() { return jsg::alloc<ProxyImpl>(); }
+
+    int getBar() { return 123; }
+
+    // NamedIntercept implementation
+    kj::Maybe<jsg::JsValue> getNamed(jsg::Lock& js, kj::StringPtr name) override {
```

Comment:
@jasnell Could we allow `getNamed()` to return an arbitrary type, which then gets JSG-wrapped?

(Most implementations would return `jsg::Optional<Something>` so that they can return none for properties that don't exist...)